src/Fmt_ast.ml

```diff
@@ -3564,15 +3564,36 @@ and fmt_structure c ctx itms =
           | _ -> false
         in
         let is_simple itm =
-          match itm.pstr_desc with
-          | Pstr_include {pincl_mod= me} | Pstr_module {pmb_expr= me} ->
+          match (c.conf.let_decl, itm.pstr_desc) with
+          | _, Pstr_include {pincl_mod= me} | _, Pstr_module {pmb_expr= me} ->
               is_simple_mod me
-          | Pstr_open _ -> true
+          | _, Pstr_open _ -> true
+          | `Compact, Pstr_value (_, bindings) ->
+              let width xe =
+                String.length
+                  (Format.asprintf "%t"
+                     (Cmts.preserve (fun () -> fmt_expression c xe)))
+              in
+              let f acc x =
+                acc && is_simple c.conf width (sub_exp ~ctx x.pvb_expr)
```

Comment:
I'm not sure that it is best to use the same `is_simple` heuristic here as is used for function args, intuitively I think it makes sense for them to be different. Perhaps try something simple like width <= 3/4 margin.